thunder/benchmarks/targets.py

```diff
@@ -168,20 +165,63 @@ def interpreter_fwd(module: Callable):
     "thunder+cudnn_layernorm+nvfuser",
 )
 
-#
-# nanogpt benchmarks
-#
+
+def get_unique_configs(config_options: Sequence[str]):
+    """
+    Get the unique configurations based on the given config options.
+
+    Args:
+        config_options: The sequence of configuration options that uniquely identify a LitGPT configuration.
+    """
+    config_names = list(sorted(c["name"] for c in configs)) if RUN_ALL_CONFIGS else IMPORTANT_CONFIGS
+    unique_config_names = {}
+    for config_name in config_names:
+        config = LitGPTConfig.from_name(config_name)
+        key = tuple(getattr(config, k) for k in config_options)
+        if config_name in IMPORTANT_CONFIGS:
+            unique_config_names[key] = config_name
+        unique_config_names.setdefault(key, config_name)
+
+    config_names = list(sorted(unique_config_names.values()))
+    return config_names
+
+
+# There are many configurations but only the following parameters affect the gelu benchmark:
+# - gelu_approximate
+# - intermediate_size
+# - block_size
+# Let's select only the configurations that differ in these parameters
+def get_configs_for_gelu():
+    return get_unique_configs(("gelu_approximate", "intermediate_size", "block_size"))
 
 
+# Sample command to run this benchmark:
+# pytest thunder/benchmarks/targets.py -k "test_litgpt_gelu" --benchmark-group-by='param:config,param:bs'
 @pytest.mark.parametrize(
     "executor,",
     executors,
     ids=executors_ids,
 )
+@pytest.mark.parametrize(
+    "bs,",
+    (
+        1,
+        2,
+    ),
+    ids=("bs1", "bs2"),
+)
 @parametrize_compute_type
-def test_nanogpt_gelu(benchmark, executor: Callable, compute_type: ComputeType):
-    gelu_bench: Benchmark = NanoGPTGeLUBenchmark(
-        config="gpt2-xl", device="cuda:0", dtype=thunder.bfloat16, requires_grad=is_requires_grad(compute_type)
+@pytest.mark.parametrize(
+    "config,",
+    get_configs_for_gelu(),
+)
+def test_litgpt_gelu(benchmark, executor: Callable, bs: int, compute_type: ComputeType, config: str):
+    gelu_bench: Benchmark = LitGPTGeluBenchmark(
+        config=config,
+        batchdims=(bs,),
+        device="cuda:0",
```

Comment:
LGTM! I just have a question here, is there a particular reason why we specify the device index here instead of leaving it to the default device?

Reply:
No, there's no special reason. It was like that before this PR, see removed line 184 https://github.com/Lightning-AI/lightning-thunder/pull/744/files#diff-118f5dfdb54f6ed69a3249fda1ba903ec309955fdea5186d1de7bc0169f929c2L184